### PR TITLE
Query builder

### PR DESCRIPTION
**Pull request by edongashi**
Also some other todos in code. Better to discuss the design idea now and leave those for later.

### CODE REVIEW

File: src/QueryTab/QueryBuilder/FilterSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  column: ColumnReference;
+};
+
+const FilterValueEditor: FunctionComponent<FilterValueEditorProps> = ({ column, query, updateQuery }) => {
+  const t = useT('QueryTab::QueryBuilder::FilterSelector::FilterValueEditor');
+
+  const filter = find(query.filters, { column });
+
+  if (!filter) {
+    return null; // Should never happen.
```
Comment (pdobacz) [inline src/QueryTab/QueryBuilder/FilterSelector.tsx:36]: maybe a warning to console, just in case?

File: src/shared-react/hooks.ts [hunk 2]
```diff
@@ -6,7 +6,18 @@ import { i18nConfig } from '../shared/constants';
 export type TFunc = TFunction<typeof i18nConfig.ns, string>;
 
 export function useT(prefix?: string): TFunc {
-  return useTranslation(i18nConfig.ns, { keyPrefix: prefix }).t;
+  const { t, i18n } = useTranslation(i18nConfig.ns, { keyPrefix: prefix });
+  return (...args: Parameters<typeof t>) => {
```
Comment (pdobacz) [inline src/shared-react/hooks.ts:10]: I know what this is for only after reading the commit title. How about a one-liner comment here?

**Review by pdobacz — APPROVED** [review]
LGTM, nice work! I just had a bunch of minor comments here

File: src/AdminTab/TableList.tsx [hunk 3]
```diff
@@ -52,30 +52,26 @@ const TableActions: FunctionComponent<TableActionsProps> = ({ table, onOpenMetab
 
   const [metabaseHintHovered, setMetabaseHintHovered] = useState(false);
   // Manage visibility manually because a programmatic tab change leaves a dangling tooltip.
-  const [examplesTooltipVisible, setExamplesTooltipVisible] = useState(false);
+  const [examplesTooltipVisible, setExamplesTooltipOpen] = useState(false);
```
[the remark this answers is not in the document]
Comment (edongashi) [reply src/AdminTab/TableList.tsx:55]: Oops, missed it - thanks. The newer ant design versions switched `visible` to `open` in a ton of components. I tried to follow suit.

File: src/QueryTab/QueryBuilder/FilterSelector.tsx (hunk 1 above)
Comment (edongashi) [reply src/QueryTab/QueryBuilder/FilterSelector.tsx:36]: I intentionally avoided prints in renders. In all other actions it prints a warning. It shouldn't happen either way. We can even throw here.

**Review by pdobacz — APPROVED** [review]

**Review by pdobacz — APPROVED** [review]

**Pull request merged**